libdnf/dnf-sack.c

```diff
@@ -107,6 +107,8 @@ dnf_sack_finalize(GObject *object)
 
     FOR_REPOS(i, repo) {
         HyRepo hrepo = repo->appdata;
+        if (!hrepo)
```

Comment:
This is [undefined behavior](https://lwn.net/Articles/342330/)...you're doing the check after dereferecing it.